src/esp/assets/ResourceManager.cpp

```diff
@@ -167,98 +167,238 @@ void ResourceManager::initPhysicsManager(
+                          false, renderLightSetup);
+
+    if (!collisionMeshSuccess) {
+      LOG(ERROR)
+          << " ResourceManager::loadScene : Scene collision mesh load failed, "
+             "Aborting scene initialization.";
+      return false;
+    }
+  }
+
+  bool semanticSceneSuccess = false;
+  // set equal to current Simulator::activeSemanticSceneID_ value
+  int activeSemanticSceneID = activeSceneIDs[1];
+  // if semantic scene load is requested and possible
+  if ((loadSemanticMesh) && (assetInfoMap.count("semantic"))) {
+    auto houseFile = sceneAttributes->getHouseFilename();
+    if (Corrade::Utility::Directory::exists(houseFile)) {
+      // check if file names exist
+      AssetInfo semanticInfo = assetInfoMap.at("semantic");
+      auto semanticSceneFilename = semanticInfo.filepath;
+      if (Corrade::Utility::Directory::exists(semanticSceneFilename)) {
+        LOG(INFO) << "ResourceManager::loadScene : Loading semantic mesh "
+                  << semanticSceneFilename;
+        activeSemanticSceneID = sceneManagerPtr->initSceneGraph();
+        bool splitSemanticMesh = sceneAttributes->getFrustrumCulling();
+
+        auto& semanticSceneGraph =
+            sceneManagerPtr->getSceneGraph(activeSemanticSceneID);
+        auto& semanticRootNode = semanticSceneGraph.getRootNode();
+        auto& semanticDrawables = semanticSceneGraph.getDrawables();
+        bool computeSemanticAABBs = splitSemanticMesh;
+        semanticSceneSuccess = loadSceneInternal(
+            semanticInfo, _physicsManager, &semanticRootNode,
+            &semanticDrawables, computeSemanticAABBs, splitSemanticMesh);
+        // regardless of load failure, original code still changed
+        // activeSemanticSceneID_
+        activeSceneIDs[1] = activeSemanticSceneID;
+        if (!semanticSceneSuccess) {
+          LOG(ERROR) << " ResourceManager::loadScene : Semantic scene mesh "
+                        "load failed.";
+        } else {
+          LOG(INFO) << "ResourceManager::loadScene : Semantic Scene Mesh : "
+                    << semanticSceneFilename << " loaded.";
+        }
+      } else {  // semantic file name does not exist but house does
+        LOG(WARNING)
+            << "ResourceManager::loadScene : Not loading semantic mesh - "
+               "File Name : "
+            << semanticSceneFilename << " does not exist.";
+      }
+    } else {  // if houseFile does not exist, ID should be activeSceneIDs[0]
+      LOG(WARNING)
+          << "ResourceManager::loadScene : Not loading semantic mesh - "
+             "File Name : "
+          << houseFile << " does not exist.";
+      activeSemanticSceneID = activeSceneIDs[0];
+    }
+  } else {  // not wanting to create semantic mesh
+    LOG(INFO) << "ResourceManager::loadScene : Not loading semantic mesh";
+  }
+  // save active semantic scene ID so that simulator can consume
+  activeSceneIDs[1] = activeSemanticSceneID;
```

Comment:
Note the logic in the above block - some assumptions were made for edge cases that the original code did not cover.  Please review.